model-server/src/main/kotlin/org/modelix/model/server/store/IgniteStoreClient.kt

```diff
@@ -103,6 +112,43 @@ class IgniteStoreClient(jdbcConfFile: File? = null, inmemory: Boolean = false) :
         return cache.associate { it.key to it.value }
     }
 
+    override fun removeRepositoryObjects(repositoryId: RepositoryId) {
+        if (!inmemory) {
+            removeRepositoryObjectsFromDatabase(repositoryId)
+        }
+
+        val filter = IgniteBiPredicate<ObjectInRepository, String?> { key, _ ->
+            key.getRepositoryId() == repositoryId.id
+        }
+        val transformer = IgniteClosure<Cache.Entry<ObjectInRepository, String?>, ObjectInRepository?> { entry ->
+            entry.key
+        }
+        val query = ScanQuery(filter)
+        val toDelete = cache.query(query, transformer).all.asSequence().filterNotNull()
+            .toSortedSet() // sorting is necessary to avoid deadlocks
```

Comment:
Why can deadlocks occur in case it's not sorted? Some reference to documentation explaining this would be good.

Reply:
added a hint